packages/@lwc/ssr-compiler/src/compile-js/lwc-import.ts

```diff
@@ -31,6 +32,14 @@ export function replaceLwcImport(path: NodePath<ImportDeclaration>, state: Compo
             state.lightningElementIdentifier = specifier.local.name;
             break;
         }
+        if (specifier.type === 'ImportSpecifier' && specifier.imported.type === 'Identifier') {
+            if (!allowedLwcImports.has(specifier.imported.name)) {
+                throw new Error(`Cannot import "${specifier.imported.name}" in SSR context.`);
+            }
+            if (specifier.imported.name === 'LightningElement') {
+                state.lightningElementIdentifier = specifier.local.name;
+            }
+        }
```

Comment:
If you can't import `LightningElement` from `'lwc'` then you shouldn't be able to export it either. Otherwise you can make a barrel file as a workaround.